### ISSUE TITLE
Add createConstraint() to Migration

### ISSUE THREAD

**Issue by YasserHassan** [yiisoft/yii2#9003]
The believe it's important to have a function for creating constraints similar to `createIndex()`.

**Comment by samdark** [comment]
Primary and foreign key constraints are already there. Do you mean something additional?

**Comment by nineinchnick** [comment]
I'm using [check and exclusion constraints](http://www.postgresql.org/docs/devel/static/ddl-constraints.html) so it would be handy to add them like indexes and fks instead directly using `execute()`.

**Comment by samdark** [comment]
Makes sense.

**Comment by nineinchnick** [comment]
Actually, FK is just a specific constraint so its syntax could be more general.

**Comment by YasserHassan** [comment]
It is important to consider that there are additional options for constraints. For example in PostgreSQL:
```
ALTER TABLE table1
ADD CONSTRAINT table1_col1_col2_key UNIQUE (col1, col2)
DEFERRABLE INITIALLY DEFERRED;
```
So the function signature could be something like this:
```
createConstraint($name, $table, $columns, $unique = false, $options = null)
```
Notice that PostgreSQL does not allow expressions in constraints so no need to check parenthesis and escape on them. I'm not sure about others databases though.

Probably it would be good for maintainability to consider letting `addForeignKey()` and `addPrimaryKey()` use `createConstraint()` internally and the same for dropping them (using `dropConstraint()`) as a future enhancement.

**Comment by YasserHassan** [comment]
A more generic signature could be:
```
createConstraint($name, $table, $columns, $type = null, $value = null, $options = null)
```
where `$type` is CHECK, UNIQUE, ...etc, and value is something like `char_length(zipcode) = 5` either as array or string. Please take a look at the constraints examples at the end of [ALTER TABLE documentation](http://www.postgresql.org/docs/9.4/static/sql-altertable.html).

**Comment by sergeymakinen** [comment]
I think this issue is now implemented in #14105.

**Comment by buttflattery** [comment]
reading the comments looks like we can do it but could not find any notes so sorry for asking how can i create a migration where i have `CHECK(column<12)` constraint like below

```
CREATE TABLE user (
ID int NOT NULL,
Age int,
CHECK (Age>=18)
);
```
